### PR TITLE
migrate kubelet --bootstrap-kubeconfig to kubelet.config.k8s.io

### LINKED ISSUE #61680: Migrate Kubelet --bootstrap-kubeconfig to kubelet.config.k8s.io or remove the flag
Flag name: `bootstrap-kubeconfig`

Help text:

Path to a kubeconfig file that will be used to get client certificate for kubelet. If the file specified by --kubeconfig does not exist, the bootstrap kubeconfig is used to request a client certificate from the API server. On success, a kubeconfig file referencing the generated client certificate and key is written to the path specified by --kubeconfig. The client certificate and key file will be stored in the directory pointed by --cert-dir.

---

This is part of migrating the Kubelet command-line to a Kubernetes-style API.

The `--bootstrap-kubeconfig` flag should either be migrated to the Kubelet's `kubelet.config.k8s.io`
API group, or simply removed from the Kubelet.

If this could be considered an instance-specific flag, or a descriptor of local topology
managed by the Kubelet, see: https://github.com/kubernetes/kubernetes/issues/61647.

If this flag is only registered in os-specific builds, see: https://github.com/kubernetes/kubernetes/issues/61649.

@sig-node-pr-reviews @sig-node-api-reviews

/assign @mtaufen
/sig node
/kind feature
/priority important-soon
/milestone v1.11
/status approved-for-milestone

### PR DESCRIPTION
**Pull request by tianshapjq**
**What this PR does / why we need it**:
migrate kubelet --boot-kubeconfig to kubelet.config.k8s.io

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #61680

**Special notes for your reviewer**:
@mtaufen 

**Release note**:

```release-note
The Kubelet's --bootstrap-kubeconfig flag can now be set via the kubelet.config.k8s.io/v1beta1 API by specifying the KubeletConfiguration.BootstrapKubeconfig field.
```

### CODE REVIEW

**Comment by mtaufen** [comment]
Need to fix the release note syntax

**Comment by mtaufen** [comment]
We should probably migrate both KubeConfig and BootstrapKubeconfig at the same time (and I'd like to rename KubeConfig to Kubeconfig when we do so).

/cc @liggitt

**Comment by ericchiang** [comment]
> on bootstrap-kubeconfig promotion to config vs extracting to an exec auth provider

To me, the bootstrap kubeconfig is about lifecycle phases. This is the user you create the initial CSR as, the real kubeconfig is the one you perform all other actions with. Your bootstrap kubeconfig could even use an exec auth provider.

I haven't thought of the exec auth provider being a replacement for the bootstrap kubeconfig, but admittedly our experiments with bootkube have only used cert rotation without this bootstrap flag.

I'd lean towards promotion to config.

**Comment by tianshapjq** [comment]
@mtaufen already migrate Kubeconfig, pls check :)

File: cmd/kubelet/app/options/options.go [hunk 1]
```diff
@@ -438,6 +427,13 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
 		mainfs.AddFlagSet(fs)
 	}()
 
+	fs.MarkDeprecated("experimental-bootstrap-kubeconfig", "Use --bootstrap-kubeconfig")
+	fs.StringVar(&c.Kubeconfig, "kubeconfig", c.Kubeconfig, "Path to a kubeconfig file, specifying how to connect to the API server. Providing --kubeconfig enables API server mode, omitting --kubeconfig enables standalone mode.")
+	fs.StringVar(&c.BootstrapKubeconfig, "experimental-bootstrap-kubeconfig", c.BootstrapKubeconfig, "deprecated: use --bootstrap-kubeconfig")
```
Comment (mtaufen) [inline cmd/kubelet/app/options/options.go]: How long has experimental-bootstrap-kubeconfig been deprecated? We can probably just remove the experimental version at this point.

**Comment by mtaufen** [comment]
/remove lifecycle-rotten

@tianshapjq sorry for the long wait, I'm finally back to working on these again.

@mikedanese @ericchiang sounds like this is ok? Or is there a plan to replace with exec auth provider?

Happy to lgtm this if we can confirm.

**Pull request closed without merging**